torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -626,6 +626,13 @@ at::Tensor XLANativeFunctions::_unsafe_view(const at::Tensor& self,
   return view_copy_symint(self, c10::fromIntArrayRefSlow(size));
 }
 
+at::Tensor XLANativeFunctions::_unsafe_index(
```

Comment:
do we know when will this method being used(how does it get dispatched to this) and create a test for it?

Reply:
+1 to adding the tests. For more info, the cpp unit tests can verify that this new op you lowered is properly invoked via metric assertions as such https://github.com/pytorch/xla/blob/master/test/cpp/test_aten_xla_tensor_1.cpp#L35.

Reply:
@peterbell10 added this function. These functions are an artefact to signal the compiler to avoid performing any boundary checks on the indices that they use. These are used in decompositions where we know that the indices will be in the right range, like the backward of functions, or a decomposition where we know that the inputs are bounded to the right range.

Reply:
Note that you also have its cousin `_unsafe_index_put`